### PR TITLE
fixing git pull comments about stream closing in the finally block.

### PR DESCRIPTION
**Pull request by swsachith**
fixing comments on stream closing in the finally block

### CODE REVIEW

File: components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java [hunk 1]
```diff
@@ -373,12 +373,16 @@ private static void backupFileSystemToLocal(AnalyticsFileSystem analyticsFileSys
...
-                int i;
                 FileOutputStream out = new FileOutputStream(baseDir+nodePath);
-                while ((i = input.read(dataInBuffer, 0, dataInBuffer.length)) > 0) {
-                    out.write(dataInBuffer);
+                while (input.read(dataInBuffer, 0, dataInBuffer.length) > 0) {
+                    try {
+                        out.write(dataInBuffer);
+                    } catch (IOException e) {
+                        throw new IOException("Could not write to the output file: ", e);
+                    }finally{
```
Comment (lafernando) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: Fix code formatting (indentation).

File: components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java [hunk 2]
```diff
@@ -418,9 +422,14 @@ private static void restoreFileStructure(AnalyticsFileSystem analyticsFileSystem
...
             byte[] data = readFile(node);
-            out.write(data, 0, data.length);
-            out.flush();
-            out.close();
+            try {
+                out.write(data, 0, data.length);
+                out.flush();
+            } catch (IOException e) {
+                throw new IOException("Error in restoring the file to the filesystem: ", e);
+            }finally{
```
Comment (lafernando) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: Fix code formatting, fix in all places.

**Comment by lankavitharana** [comment]
Seems the catch block is redundant as the same exception is thrown later.

File: components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java (hunk 2 above)
Comment (swsachith) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: These have been fixed in a later PRs.

File: components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java (hunk 1 above)
Comment (swsachith) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: These have been fixed in a later PRs.

**Comment by swsachith** [comment]
I wanted to add a custom message to the exception.

**Pull request merged**